homeassistant/components/lifx/light.py

```diff
@@ -97,7 +98,10 @@ async def async_setup_entry(
         "set_hev_cycle_state",
     )
     if lifx_features(device)["matrix"]:
-        entity: LIFXLight = LIFXMatrix(coordinator, manager, entry)
+        if device.product in {176, 177}:
```

Comment:
Would be nice if 176/177 was a named constant from the lib or a function to tell its a ceiling light

Reply:
Yes, it would. I've asked LIFX to add this to their official products.json file, but in the meantime, there is no other programmatic way to determine what is (or is not) a Ceiling device besides the product ID. I could put this in `aiolifx`, I guess while I continue to push LIFX to improve their product definitions.

Reply:
> I could put this in `aiolifx`, I guess while I continue to push LIFX to improve their product definitions.

That would be a great solution for this case 👍

Reply:
It would be a great solution for Home Assistant, but I'm not sure it's a great solution for `aiolifx`, to be honest. I don't want yet another technical debt because we've hacked something in before LIFX makes it part of the official spec, then I can't roll it back. I'd rather just do it here as a workaround so that other dependents of `aiolifx` don't use it.

Reply:
It looks like `update_products` is already adding/converting keys in https://github.com/aiolifx/aiolifx/blob/54c3c0f5533c532ee9709c2bee722e6f120797ba/aiolifx/update_products.py#L31

Maybe  `ceiling` or similar could be added in https://github.com/aiolifx/aiolifx/blob/54c3c0f5533c532ee9709c2bee722e6f120797ba/aiolifx/products_defs.py#L131

Reply:
It's not adding or changing any keys. It's just creating a single more easily usable dictionary from the upstream production definitions: https://github.com/LIFX/products/blob/master/products.json